File: 345.reverse-vowels-of-a-string.cs
/*
 * @lc app=leetcode id=345 lang=csharp
 *
 * [345] Reverse Vowels of a String
 */

// @lc code=start
public class Solution {
    public string ReverseVowels(string s) {
        HashSet<char> vowels = new HashSet<char>();
        string Vowels = "aeiouAEIOU";
        foreach (char c in Vowels)
        {
            vowels.Add(c);
        }
        int left = 0;
        int right = s.Length - 1;
        char[] arr = s.ToCharArray();
        while (left < right)
        {
            if (!vowels.Contains(arr[left]))
            {
                left += 1;
            }
            else if (!vowels.Contains(arr[right]))
            {
                right -= 1;
            }
            else
            {
                char temp = arr[left];
                arr[left] = arr[right];
                arr[right] = temp;
                left += 1;
                right -= 1;
            }
        }
        return new string(arr);
    }
}
// @lc code=end

File: 345.reverse-vowels-of-a-string.py
#
# @lc app=leetcode id=345 lang=python3
#
# [345] Reverse Vowels of a String
#

# @lc code=start
class Solution:
    def reverseVowels(self, s: str) -> str:
        string_list = []
        for c in s:
            string_list.append(c)
        vowels = "aeiouAEIOU"
        left, right = 0, len(string_list)-1
        while left < right:
            if string_list[left] not in vowels:
                left += 1
            elif string_list[right] not in vowels:
                right -= 1
            else:
                string_list[left], string_list[right] = string_list[right], string_list[left]
                left += 1
                right -= 1
        return "".join(string_list)
        
# @lc code=end

